components/builder-depot/src/server.rs

```diff
@@ -455,6 +457,36 @@ fn write_archive(filename: &PathBuf, body: &mut Body) -> Result<PackageArchive>
     Ok(PackageArchive::new(filename))
 }
 
+fn download_latest_origin_encryption_key(req: &mut Request) -> IronResult<Response> {
+    let origin = match get_param(req, "origin") {
+        Some(origin) => origin,
+        None => return Ok(Response::with(status::BadRequest)),
+    };
+
+    if !check_origin_access(req, &origin).unwrap_or(false) {
+        return Ok(Response::with(status::Forbidden));
+    }
+
+    let mut request = OriginPublicEncryptionKeyLatestGet::new();
+    match helpers::get_origin(req, origin) {
+        Ok(mut origin) => {
+            request.set_owner_id(origin.get_owner_id());
+            request.set_origin(origin.take_name());
+        }
+        Err(err) => return Ok(render_net_error(&err)),
+    }
+    let key = match route_message::<
+        OriginPublicEncryptionKeyLatestGet,
+        OriginPublicEncryptionKey,
+    >(req, &request) {
+        Ok(key) => key,
+        Err(err) => return Ok(render_net_error(&err)),
+    };
+
+    let xfilename = format!("{}-{}.pub", key.get_name(), key.get_revision());
```

Comment:
might not be a bad idea extracting a `public_key_file_name` function since we are using this format in a few places in this file.

Reply:
I tried putting this logic in `download_content_as_file` but there is `sig.key` and `pub` types which added more checking logic than it removed